.gitlab-ci.yml

```diff
@@ -0,0 +1,87 @@
+
+# /************************************************************************
+ #  File: .gitlab-ci.yml
+ #  Author: mdr0id
+ #  Date: 9/10/2018
+ #  Description:  Used to setup runners/jobs for librustzcash
+ #  Usage: Commit source and the pipeline will trigger the according jobs.
+ #
+ #  Known bugs/missing features:
+ #
+ # ************************************************************************/
+  
+stages:
+  - build
+  - test
+  - deploy
+
+rust-latest:
+  stage: build
+  image: rust:latest
+  script:
+    - cargo --verbose --version
+    - rustc --verbose --version
+    - time cargo build --verbose
+
+#Ignore the nightly debug CI job for now (9/18/2018)
+.rust-debug-latest:
+  stage: build
+  image: rust:latest
+  script:
+    - cargo build --debug --verbose
+
+rust-nightly:
+  stage: build
+  image: rustlang/rust:nightly
+  script:
+    - cargo --verbose --version
+    - rustc --verbose --version
+    - cargo build --verbose
+  allow_failure: true
+
+#Ignore the nightly debug CI job for now (9/18/2018)
+.rust-debug-nightly:
+  stage: build
+  image: rustlang/rust:nightly
+  script:
+    - cargo build --debug --verbose
+  allow_failure: true
+
+librustzcash-test-latest:
+  stage: test
+  image: rust:latest
+  script:
+    - cargo --verbose --version
+    - rustc --verbose --version
+    - time cargo test --release --verbose
+
+librustzcash-test-rust-nightly:
+  stage: test
+  image: rustlang/rust:nightly
+  script:
+    - cargo --verbose --version
+    - rustc --verbose --version
+    - cargo test --release --verbose
+  allow_failure: true
+
+librustzcash-rust-rc:
+  stage: deploy
+  image: rust:latest
+  script:
+    - cargo --verbose --version
+    - rustc --verbose --version
+    - time cargo build --release --verbose
+    - time cargo test --release --verbose
+    - cargo doc --verbose
```

Comment:
This makes me wonder whether the stages are in fact completely independent. If they aren't, then the `build` and `test` lines above are unnecessary (and should do nothing). If they are, then I don't see the benefit of the earlier stages, as the test suite would be built and run twice.

Reply:
Disclaimer: I am rather new to Rust, but after looking at the logs it looks like test does a build too. I have removed the redundant commands.